app/process/watcher/publisher/publish.go

```diff
@@ -18,19 +18,11 @@ package publisher
 
 import (
 	"github.com/golang/protobuf/proto"
-	"github.com/limechain/hedera-watcher-sdk/queue"
-	"github.com/limechain/hedera-watcher-sdk/types"
-	log "github.com/sirupsen/logrus"
+	"github.com/limechain/hedera-eth-bridge-validator/app/core/pair"
 )
 
-func Publish(m proto.Message, typeMessage string, id interface{}, q *queue.Queue) {
-	message, e := proto.Marshal(m)
-	if e != nil {
-		log.Fatalf("[%s] - Failed marshalling response - ID: [%s]\n", typeMessage, id)
-	}
-
-	q.Push(&types.Message{
-		Payload: message,
-		Type:    typeMessage,
+func Publish(m proto.Message, q *pair.Queue) {
+	q.Push(&pair.Message{
```

Comment:
now that the publisher is so "thin" and only 1 line of code is the actual publishing, wouldn't it be better to remove it and call directly `q.Push` from the watchers.